src/azure-cli-core/azure/cli/core/cloud.py

```diff
@@ -0,0 +1,181 @@
+#---------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+#---------------------------------------------------------------------------------------------
+
+import os
+from six.moves import configparser
+
+from azure.cli.core._config import GLOBAL_CONFIG_DIR
+
+CUSTOM_CLOUD_CONFIG_FILE = os.path.join(GLOBAL_CONFIG_DIR, 'clouds.config')
+
+class CloudNotRegisteredException(Exception):
+    def __init__(self, cloud_name):
+        super(CloudNotRegisteredException, self).__init__(cloud_name)
+        self.cloud_name = cloud_name
+    def __str__(self):
+        return "The cloud '{}' is not registered.".format(self.cloud_name)
+
+class CloudAlreadyRegisteredException(Exception):
+    def __init__(self, cloud_name):
+        super(CloudAlreadyRegisteredException, self).__init__(cloud_name)
+        self.cloud_name = cloud_name
+    def __str__(self):
+        return "The cloud '{}' is already registered.".format(self.cloud_name)
+
+class CannotUnregisterCloudException(Exception):
+    pass
+
+class CloudEndpointUrl(object): # pylint: disable=too-few-public-methods
+    MANAGEMENT = 'management'
```

Comment:
I think @BurtBiel would suggest that these be lower cased since ultimately they're just properties and not really constants. For ease of creating Azure Stack endpoint objects, I would recommend you give these objects a constructor so you can do this:

```Python
myendpoint = CloudEndpointUrl(
  'managment.wonky.com',
  'sql.wonky.com',
  ...)
```
instead of:
```Python
myendpoint = CloudEndpointUrl()
myendpoint.MANAGEMENT = 'managment.wonky.com'
myendpoint.SQL_MANAGEMENT = 'sql.wonky.com'
```

The all caps in the second version are even more strange because it looks like you are setting the value of a constant (which of course they aren't really constants)

Finally, the init would allow you to default certain endpoints (perhaps to public Azure?) if you wanted to.